### PR TITLE
Ports Mapping Backend updates

### PR DESCRIPTION
**Pull request by fira**
# About the pull request

Updates most of mapping backend to /tg/ current:
* Replaces completely map reader which now has explicit TGM support
* Replaces completely reservations backend, they now have a builtin "cordon" area around them
* Grafts and updates some missing procs in SSmapping and Shuttles, notably for reservations
* Updates nightmare, map templates, interiors, etc, to match new mapper/reservations code
* Minor Turf/Atoms stuff such as global init, turf_flags rename, and AfterChange()

This shouldn't change much to the game as is, but is meant as a stepping stone toward new features such as planecube that depend on updated mapping code, rather than having to messily strap them to old.

Very Testmerge-worth

# Testing
 - [x] General map loading of Ship & Ground
 - [x] Dropship loading and flight
 - [x] Fultons reservation launching
 - [x] Lifeboats launching
 - [x] Vehicle Interiors instanciation
 - [x] Interior reservations freeing
 - [x] Tents templates placement
 - [x] Nightmare inserts loading
 - [x] Other maps than LV-624
 - [x] Maploading admin verbs
 - [x] Mass screenshot verb
 - [x] Runtime Trijent elevators working

# Potential TODO
 - [x] Find a better cordon turf than outer almayer hull, it looks ugly with eg fultons

# Changelog
:cl:
code: Updated mapping backend from /tg/ upstream.
/:cl:

### CODE REVIEW

**Comment by fira** [comment]
Currently removes Weakrefs for used_turfs reservations because they weren't in /tg/ backend

This doesn't seem to trip C&D so.... I don't know

**Comment by Huffie56** [comment]
could it be fixing those two issues?
https://github.com/cmss13-devs/cmss13/issues/2798
https://github.com/cmss13-devs/cmss13/issues/2799

**Comment by fira** [comment]
#2798 is probably rather an issue with the verb that should our TGMC/CM snowflake delete argument for this, we previously had this issue with nightmare

#2799 could be an old icon issue that happened on old server due to docker container not packing icons

Both are very old issues so hard to tell

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by mullenpaul** [comment]
I'll review this later - I need mapping updated for some of my projects

File: code/modules/shuttle/shuttles/trijent_elevator.dm [hunk 1]
```diff
@@ -37,12 +37,6 @@
 	. = ..()
 	door_control.control_doors("force-lock-launch", "all", force=TRUE)
 
-/obj/docking_port/mobile/trijent_elevator/linkup(datum/map_template/shuttle/template, obj/docking_port/stationary/dock)
```
Comment (mullenpaul) [inline code/modules/shuttle/shuttles/trijent_elevator.dm:40]: I'm not sure if this should be removed. Was it moved to another proc?

**Review by mullenpaul — APPROVED** [review]
I've had a look at the code and it generally looks fine. There is a LOT here - how has this been tested? We should run a TM at some point

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by dadlips** [comment]
I believe this TM has introduced a bug with the dropships where the aft door stays LOCKED, and can't be unlocked from any console. Forest mentioned the dropships seem to think they're in space all the time, and when one took off I did see some space tile oddities I hadn't before.

**Comment by Vicacrov** [comment]
Tested it on local (start round, simply spawn in):

1. South dropship doors are locked by default, cannot be unlocked
2. The corners of the dropships have space tiles

image

**Review by Drulikar — CHANGES_REQUESTED** [review]
Please address issues with the dropships (but I believe earlier comments also isolated it to soley this PR)

image

Nothing immediately sticks out to me as the issue, but if you can't figure it out I can dig into it more later.

File: code/game/world.dm [hunk 2]
```diff
@@ -321,9 +323,39 @@ GLOBAL_LIST_INIT(reboot_sfx, file2list("config/reboot_sfx.txt"))
...
+		return
+//	var/area/global_area = GLOB.areas_by_type[world.area] // We're guarenteed to be touching the global area, so we'll just do this
+//	var/list/to_add = block(
+//		locate(1, old_maxy + 1, 1),
+//		locate(maxx, maxy, map_load_z_cutoff))
+//	global_area.contained_turfs += to_add
+
 /world/proc/incrementMaxZ()
 	maxz++
-	//SSmobs.MaxZChanged()
```
Comment (Drulikar) [inline code/game/world.dm:359]: These procs aren't updating the zlevel's bounds resulting in those values remaining as 100x150x1 resulting in bogus values for mass screenshot.

**Review by spookydonut — APPROVED** [review]

**Review by Drulikar — CHANGES_REQUESTED** [review]
Can also confirm issue 5504 is from this PR on big red (atleast on 515)
image

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by harryob — APPROVED** [review]

**Pull request merged**